src/FilesystemManagement.cpp

```diff
@@ -0,0 +1,365 @@
+                 ConvertErrNo(errno) );
+    #endif // MEZZ_Windows
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Symlinks
+
+    ModifyResult CreateSymlink(const StringView SymPath, const StringView TargetPath)
+    {
+    #ifdef MEZZ_Windows
+        DWORD LinkFlags = 0;
+        std::wstring WideSymPath = ConvertToWideString(SymPath);
+        std::wstring WideTargetPath = ConvertToWideString(TargetPath);
+        return ( ::CreateSymbolicLinkW(WideSymPath.data(),WideTargetPath.data(),LinkFlags) != 0 ?
+                 ModifyResult::Success :
+                 ConvertErrNo( ::GetLastError() ) );
+    #else // MEZZ_Windows
+        return ( ::symlink(SymPath.data(),TargetPath.data()) == 0 ?
+                 ModifyResult::Success :
+                 ConvertErrNo(errno) );
+    #endif // MEZZ_Windows
+    }
+
+    ModifyResult CreateDirectorySymlink(const StringView SymPath, const StringView TargetPath)
+    {
+    #ifdef MEZZ_Windows
+        DWORD LinkFlags = SYMBOLIC_LINK_FLAG_DIRECTORY;
+        std::wstring WideSymPath = ConvertToWideString(SymPath);
+        std::wstring WideTargetPath = ConvertToWideString(TargetPath);
+        return ( ::CreateSymbolicLinkW(WideSymPath.data(),WideTargetPath.data(),LinkFlags) != 0 ?
+                 ModifyResult::Success :
+                 ConvertErrNo( ::GetLastError() ) );
+    #else // MEZZ_Windows
+        return ( ::symlink(SymPath.data(),TargetPath.data()) == 0 ?
+                 ModifyResult::Success :
+                 ConvertErrNo(errno) );
+    #endif // MEZZ_Windows
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Basic Directory Management
+
+    Boole DirectoryExists(const StringView DirectoryPath)
+    {
+    #ifdef MEZZ_Windows
+        std::wstring WidePath = ConvertToWideString(DirectoryPath);
+        DWORD dwAttrib = GetFileAttributesW(WidePath.data());
+        return ( dwAttrib != INVALID_FILE_ATTRIBUTES && ( dwAttrib & FILE_ATTRIBUTE_DIRECTORY ) );
+    #else // MEZZ_Windows
+        struct stat st;
+        if( stat(DirectoryPath.data(),&st) == 0 ) {
+            return S_ISDIR(st.st_mode);
+        }
+        return false;
+    #endif // MEZZ_Windows
+    }
+
+    ModifyResult CreateDirectory(const StringView DirectoryPath)
+    {
+    #ifdef MEZZ_Windows
+        std::wstring WidePath = ConvertToWideString(DirectoryPath);
+        return ( ::CreateDirectoryW(WidePath.data(),NULL) != 0 ?
+                 ModifyResult::Success :
+                 ConvertErrNo( ::GetLastError() ) );
+    #else // MEZZ_Windows
+        return ( ::mkdir(DirectoryPath.data(),0755) == 0 ?
+                 ModifyResult::Success :
+                 ConvertErrNo(errno) );
+    #endif // MEZZ_Windows
+    }
+
+    ModifyResult CreateDirectoryPath(const StringView DirectoryPath)
+    {
+        ModifyResult Result = ModifyResult::Success;
+    #ifdef MEZZ_Windows
```

Comment:
These first two ifdefs could be consolidated with some fairly direct refactorizations. VecIndex, PathAttempt and SysSlash could be moved above the first then the rest of the code combined. I think some vertical whitespace would be nice around these, but I suspect you will disagree on that part.

Reply:
I generally do like one vertical white space between variables being created at the start and the start of the logic.  Refactored.